### PR TITLE
Drop 'brodocs' backend for API reference generation

### PR DESCRIPTION
**Pull request by tengqm**
We cannot afford maintaining two backends for generating the reference
docs. This PR drops the legacy 'brodocs' way of building refernece docs.
We drop it because:

- The brodocs container has to be managed by a specific person so the
  team collaboration model is not scalable.
- The brodocs container has not been maintained for a long period.
- The markdown output from the brodocs container are difficult to tune
  for web presentation -- main use case of the tool.
- The new 'html' generator has been used for several release cycles and
  it is generating satisfactory output for users. It can be considered
  mature.

### CODE REVIEW

File: gen-apidocs/generators/writer.go [hunk 1]
```diff
@@ -62,15 +58,7 @@ func GenerateFiles() {
...
-	var writer DocWriter
-	if *Backend == "brodocs" {
-		writer = NewMarkdownWriter(config, copyright, title)
-	} else if *Backend == "go" {
-		writer = NewHTMLWriter(config, copyright, title)
-	} else {
-		panic(fmt.Sprintf("Unknown backend specified: %s", *Backend))
-	}
-
+    writer := NewHTMLWriter(config, copyright, title)
```
Comment (kbhawkey) [inline gen-apidocs/generators/writer.go]: nit: indentation of line 61?

File: gen-apidocs/generators/util.go [hunk 2]
```diff
@@ -84,74 +84,3 @@ func PrintInfo(config *api.Config) {
 	//	}
 	//}
 }
-
```
Comment (kbhawkey) [inline gen-apidocs/generators/util.go]: Is PrintInfo(), JsonOutputFile(manifest.json) used?

File: gen-apidocs/generators/writer.go (hunk 1 above)
Comment (tengqm) [reply gen-apidocs/generators/writer.go]: will check

Comment (tengqm) [reply gen-apidocs/generators/writer.go]: indentation should be one TAB according Go convention. I have got TABs autoconverted to 4 spaces in my VIM. Will fix.

File: gen-apidocs/generators/util.go (hunk 2 above)
Comment (tengqm) [reply gen-apidocs/generators/util.go]: PrintInfo is still referenced, by JsonOutputFile is not (will fix).

Comment (kbhawkey) [reply gen-apidocs/generators/util.go]: @tengqm , How do these changes affect the generation of the kubectl command reference?

```
docker run -v $(shell pwd)/gen-kubectldocs/generators/includes:/source -v $(shell pwd)/gen-kubectldocs/generators/build:/build -v $(shell pwd)/gen-kubectldocs/generators/:/manifest pwittrock/brodocs
```
Eventually the tooling/build of the kubectl reference will be updated (I hope to reopen the related kubectl reference generation PR and continue work) too.

Comment (tengqm) [reply gen-apidocs/generators/util.go]: This change is irrelevant to the gen-kubectldocs. This PR is only related to the API reference generation. I'd like to have the kubectl reference reworked as well.

**Comment by kbhawkey** [comment]
@tengqm , Looks fine. Do you know what version of the k8s API was the last version to be generated from the `brodocs` backend?

**Comment by tengqm** [comment]
> @tengqm , Looks fine. Do you know what version of the k8s API was the last version to be generated from the `brodocs` backend?

The "new" generator was merged in July 2018 (https://github.com/kubernetes-sigs/reference-docs/pull/53). So IIRC, 1.11 was the last version that used `brodocs` for API reference generation.

**Pull request merged**